python/sqlflow_submitter/tensorflow/diag.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2020 The SQLFlow Authors. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import re
+
+
+class SQLFlowDiagnostic(Exception):
```

Comment:
Exception names should be like `XXXError` and `XXXException`?

Reply:
From https://www.python.org/dev/peps/pep-0008/#exception-names

> Because exceptions should be classes, the class naming convention applies here. However, you should use the suffix "Error" on your exception names (if the exception actually is an error).

Maybe `Diagnostic` is more accurate.

Reply:
It's indeed some error I think? When some error occurs, we should analyze the error and return useful messages, and this process is "Diagnostic".

Reply:
I see what you mean, `raise an SQLFlowDiagnositic Error ` is more accurate.